pilot/pkg/config/coredatamodel/discovery.go

```diff
@@ -53,34 +51,32 @@ func (o *DiscoveryOptions) mixerEnabled() bool {
 	return o.Env != nil && o.Env.Mesh != nil && (o.Env.Mesh.MixerCheckServer != "" || o.Env.Mesh.MixerReportServer != "")
 }
 
-// MCPDiscovery provides storage for NotReadyEndpoints
+// MCPDiscovery provides discovery interface for SyntheticServiceEntries
 type MCPDiscovery struct {
-	notReadyEndpointsMu sync.RWMutex
-	// [ip:port]config
-	notReadyEndpoints map[string]*model.Config
+	*SyntheticServiceEntryController
 	*DiscoveryOptions
 }
 
 // NewMCPDiscovery provides a new instance of Discovery
-func NewMCPDiscovery(options *DiscoveryOptions) *MCPDiscovery {
+func NewMCPDiscovery(controller CoreDataModel, options *DiscoveryOptions) *MCPDiscovery {
 	return &MCPDiscovery{
-		notReadyEndpoints: make(map[string]*model.Config),
-		DiscoveryOptions:  options,
+		SyntheticServiceEntryController: controller.(*SyntheticServiceEntryController),
```

Comment:
nit: could you just make the argument of type `*SyntheticServiceEntryController`?

Reply:
That would still require type assertion unless I change the return type on the controller itself (from Coredatamodel to *SyntheticServiceEntryController), however I was hoping to avoid that since I wanted to keep the contract on the controller aligned and consistent with other MCP controller.